include/keyboard.h

```diff
@@ -99,6 +99,151 @@ enum KBD_KEYS {
 	// clang-format on
 };
 
+enum ScanCode : uint8_t {
+	SC_None,
+	SC_AltEscape,
+	SC_AltSpace,
+	SC_ControlInsert = 0x04,
+	SC_ShiftInsert,
+	SC_ControlDelete,
+	SC_ShiftDelete,
+	SC_AltBackspace,
+	SC_AltShiftBackspace,
+	SC_ShiftTab = 0x0F,
+	SC_AltQ,
+	SC_AltW,
+	SC_AltE,
+	SC_AltR,
+	SC_AltT,
+	SC_AltY,
+	SC_AltU,
+	SC_AltI,
+	SC_AltO,
+	SC_AltP,
+	SC_AltOpenBracket,
+	SC_AltCloseBracket,
+	SC_AltA = 0x1E,
+	SC_AltS,
+	SC_AltD,
+	SC_AltF,
+	SC_AltG,
+	SC_AltH,
+	SC_AltJ,
+	SC_AltK,
+	SC_AltL,
+	SC_AltSemicolon,
+	SC_AltApostrophe,
+	SC_AltBacktick,
+	SC_AltBackslash = 0x2B,
+	SC_AltZ,
+	SC_AltX,
+	SC_AltC,
+	SC_AltV,
+	SC_AltB,
+	SC_AltN,
+	SC_AltM,
+	SC_AltComma,
+	SC_AltPeriod,
+	SC_AltSlash,
+	SC_AltNumpadAsterisk = 0x37,
+	SC_F1                = 0x3B,
+	SC_F2,
+	SC_F3,
+	SC_F4,
+	SC_F5,
+	SC_F6,
+	SC_F7,
+	SC_F8,
+	SC_F9,
+	SC_F10,
+	SC_Home = 0x47,
+	SC_Up,
+	SC_PageUp,
+	SC_Left = 0x4B,
+	SC_Center,
+	SC_Right,
+	SC_AltPlus,
+	SC_End,
+	SC_Down,
+	SC_PageDown,
+	SC_Insert,
+	SC_Delete,
+	SC_ShiftF1,
+	SC_ShiftF2,
+	SC_ShiftF3,
+	SC_ShiftF4,
+	SC_ShiftF5,
+	SC_ShiftF6,
+	SC_ShiftF7,
+	SC_ShiftF8,
+	SC_ShiftF9,
+	SC_ShiftF10,
+	SC_ControlF1,
+	SC_ControlF2,
+	SC_ControlF3,
+	SC_ControlF4,
+	SC_ControlF5,
+	SC_ControlF6,
+	SC_ControlF7,
+	SC_ControlF8,
+	SC_ControlF9,
+	SC_ControlF10,
+	SC_AltF1,
+	SC_AltF2,
+	SC_AltF3,
+	SC_AltF4,
+	SC_AltF5,
+	SC_AltF6,
+	SC_AltF7,
+	SC_AltF8,
+	SC_AltF9,
+	SC_AltF10,
+	SC_ControlPrintScreen,
+	SC_ControlLeft,
+	SC_ControlRight,
+	SC_ControlEnd,
+	SC_ControlPageDown,
+	SC_ControlHome,
+	SC_Alt1,
+	SC_Alt2,
+	SC_Alt3,
+	SC_Alt4,
+	SC_Alt5,
+	SC_Alt6,
+	SC_Alt7,
+	SC_Alt8,
+	SC_Alt9,
+	SC_Alt0,
+	SC_AltMinus,
+	SC_AltEquals,
+	SC_ControlPageUp,
+	SC_F11,
+	SC_F12,
+	SC_ShiftF11,
+	SC_ShiftF12,
+	SC_ControlF11,
+	SC_ControlF12,
+	SC_AltF11,
+	SC_AltF12,
+	SC_ControlUp,
+	SC_ControlMinus,
+	SC_ControlCenter,
+	SC_ControlPlus,
+	SC_ControlDown,
+	SC_ControlTab = 0x94,
+	SC_AltHome    = 0x97,
+	SC_AltUp,
+	SC_AltPageUp = 0x99,
+	SC_AltLeft   = 0x9B,
+	SC_AltRight  = 0x9D,
+	SC_AltEnd    = 0x9F,
+	SC_AltDown,
+	SC_AltPageDown,
+	SC_AltInsert,
+	SC_AltDelete,
+	SC_AltTab = 0xA5,
+};
```

Comment:
Very nice! Is it possible to turn it into an enum class, though, then drop the `SC_` prefix? Then we can use proper enum scoping, e.g. `ScanCode::AltTab`.

Reply:
Having said that, sometimes it's just more convenient to use namespaces instead (although we avoid namespaces in general, but I think in this very particular case they add value).

I've done this in `midi.h` where I just wanted nicely scoped constants (e.g. `MidiStatus::NoteOff`) but without the hassle of having to cast them to `uint8_t` all the time. Plus arguably these are not _really_ enums, in the sense that exhaustiveness-checking is basically meaningless in pretty much all use-cases.

TL;DR consider `namespace` too and use your judgement 😎 🤘🏻 

```cpp
namespace MidiStatus {
// Channel Voice Messages -- lower 4-bits (nibble) specify one of the 16 MIDI
// Channels (channel 1 = 0x0, channel 16 = 0xf)
constexpr uint8_t NoteOff         = 0x80;
constexpr uint8_t NoteOn          = 0x90;
constexpr uint8_t PolyKeyPressure = 0xa0;
constexpr uint8_t ControlChange   = 0xb0;
constexpr uint8_t ProgramChange   = 0xc0;
constexpr uint8_t ChannelPressure = 0xd0;
constexpr uint8_t PitchBend       = 0xe0;
...
```